apps/backend/src/modules/activity-logs/activity-log.service.ts

```diff
@@ -150,11 +150,13 @@ class ActivityLogService implements Service {
 		const contributorMap: Record<string, number[]> = {};
 
 		for (const contributor of allContributors.items) {
-			const uniqueKey = `${contributor.name}_${String(contributor.id)}`;
-			contributorMap[uniqueKey] = Array.from(
-				{ length: dateRange.length },
-				() => INITIAL_COMMITS_NUMBER,
-			);
+			if (!contributor.isHidden) {
+				const uniqueKey = `${contributor.name}_${String(contributor.id)}`;
+				contributorMap[uniqueKey] = Array.from(
```

Comment:
will look into this further and try to remove the check

Reply:
So what's the issue with adding it as where condition to the query?

Reply:
The query works fine, as we only retrieve logs from contributors who are not hidden. However, in the service, we add all contributors to the contributorMap, and at the end, we include this contributors also, but 0 logs. So, I think this check should remain.

Reply:
I think instead we need to pass option to findAll contributors, like - hasHidden, same as we did for hasDeleted

Reply:
If I understand correctly, you’re proposing to also filter contributors in the findAll method. While this approach will work, it will also apply the filtering directly to the contributors table

Reply:
for the contributors table we need to pass this option as true, so nothing should be changed in that functionality

Reply:
hope I got, thanks for patience :)